pull/github.go

```diff
@@ -269,13 +269,40 @@ func (ghc *GitHubContext) Reviews() ([]*Review, error) {
 
 func (ghc *GitHubContext) RepositoryCollaborators() (map[string]string, error) {
 	if ghc.collaborators == nil {
-		if err := ghc.loadCollaboratorData(); err != nil {
-			return nil, err
+		opts := github.ListCollaboratorsOptions{
+			ListOptions: github.ListOptions{
+				PerPage: 100,
+			},
+		}
+
+		collaborators := make(map[string]string)
+		for {
+			users, res, err := ghc.client.Repositories.ListCollaborators(ghc.ctx, ghc.owner, ghc.repo, &opts)
+			if err != nil {
+				return nil, errors.Wrap(err, "failed to list repository collaborators")
+			}
+			for _, u := range users {
+				collaborators[u.GetLogin()] = coalescePermission(u.GetPermissions())
+			}
+			if res.NextPage == 0 {
+				break
+			}
+			opts.Page = res.NextPage
 		}
+		ghc.collaborators = collaborators
 	}
 	return ghc.collaborators, nil
 }
 
+func coalescePermission(perms map[string]bool) string {
+	for _, p := range []string{"admin", "push", "pull"} {
```

Comment:
While reviewing this, I think I discovered a separate bug for `WriteCollaborators` where we assume the permission for writing is `write` rather than `push`

https://github.com/palantir/policy-bot/blob/develop/policy/common/actor.go#L39

Reply:
I think `write` is correct there, since it uses a different API: https://docs.github.com/en/rest/reference/repos#get-repository-permissions-for-a-user

I think the bug is actually in my conversion... the GraphQL object I replaced uses `read` and `write`: https://docs.github.com/en/graphql/reference/enums#repositorypermission